### PR TITLE
Relax WatchSemanticsTest to make it faster

### PR DESCRIPTION
**Pull request by wojtek-t**
Ref https://github.com/kubernetes/kubernetes/issues/123850

/kind flake
/sig api-machinery
/priority important-soon

```release-note
NONE
```

/assign @liggitt @serathius

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/storage/testing/watcher_tests.go [hunk 1]
```diff
@@ -1492,7 +1492,6 @@ func RunWatchSemantics(ctx context.Context, t *testing.T, store storage.Interfac
 				createdPods = append(createdPods, out)
 			}
 			testCheckResultsInStrictOrder(t, w, scenario.expectedEventsAfterEstablishingWatch(createdPods))
-			testCheckNoMoreResults(t, w)
```
Comment (wojtek-t) [inline staging/src/k8s.io/apiserver/pkg/storage/testing/watcher_tests.go:1495]: This sleeps 1s and with 18 testcases, it just means 18s.

I don't think it adds visible value to have it so I would just remove it for now to get some space here.

Comment (liggitt) [reply staging/src/k8s.io/apiserver/pkg/storage/testing/watcher_tests.go:1495]: I would keep it but cut the wait time waaaay down (50ms or 100ms)

Comment (wojtek-t) [reply staging/src/k8s.io/apiserver/pkg/storage/testing/watcher_tests.go]: Good idea - fixed.

**Comment by wojtek-t** [comment]
@liggitt - PTAL

**Comment by wojtek-t** [comment]
Just fixed gofmt

**Pull request merged**